spec/controllers/attachments_controller_spec.rb

```diff
@@ -96,34 +109,151 @@
+      let(:updated_attachment) do
+        {
+          file: updated_file,
+          title: "updated test attachment upload",
+        }
+      end
 
-      expect(document.attachments.count).to eq(2)
-      expect(response).to redirect_to(edit_document_path(document_type_slug: document_type_slug, content_id_and_locale: "#{document_content_id}:#{document_locale}"))
+      it "redirects to the specalist document edit page" do
+        allow_any_instance_of(AttachmentsController).to receive(:fetch_document).and_return(document)
+
+        post :update, params: {
+          document_type_slug: document_type_slug,
+          document_content_id_and_locale: "#{document_content_id}:#{document_locale}",
+          attachment_content_id: attachment_content_id,
+          attachment: updated_attachment,
+        }
+
+        expect(document.attachments.count).to eq(2)
+        expect(response).to redirect_to(edit_document_path(
+                                          document_type_slug: document_type_slug,
+                                          content_id_and_locale: "#{document_content_id}:#{document_locale}",
+                                        ))
+      end
+
+      context "update_attachment fails" do
+        it "shows an error message and redirects to edit document attachment page" do
+          allow_any_instance_of(AttachmentsController).to receive(:fetch_document).and_return(document)
+          allow_any_instance_of(Document).to receive(:update_attachment).and_return(false)
+          error_message = "There was an error updating the attachment, please try again later."
+
+          post :update, params: {
+            document_type_slug: document_type_slug,
+            document_content_id_and_locale: "#{document_content_id}:#{document_locale}",
+            attachment_content_id: attachment_content_id,
+            attachment: updated_attachment,
+          }
+
+          expect(flash[:danger]).to eq(error_message)
+          expect(response).to redirect_to(
+            edit_document_attachment_path(
+              document_type_slug: document_type_slug,
+              document_content_id_and_locale: "#{document_content_id}:#{document_locale}",
+              attachment_content_id: attachment_content_id,
+            ),
+          )
+        end
+      end
+    end
+
+    context "only the attachment title is updated" do
+      let(:document) { CmaCase.find(document_content_id, document_locale) }
+      let(:updated_file) { Rack::Test::UploadedFile.new("spec/support/images/updated_cma_case_image.jpg", "mime/type") }
+      let(:updated_attachment) do
+        {
+          title: "updated test attachment upload",
+        }
+      end
+
+      it "updates the attachment title" do
+        allow_any_instance_of(AttachmentsController).to receive(:fetch_document).and_return(document)
+        success_message = "Attachment successfully updated"
+
+        post :update, params: {
```

Comment:
this isn't anything to do with you, but generally updates use should be PATCH'ing rather than POST'ing. Although, i'm not sure we need to worry about it in this PR as we're just adding coverage.

Reply:
Ah yes, I only used post as that was used previously and didn't think to check. I can change all of the update tests to patch requests

Reply:
Honestly, it's nothing to do with the scope of the card so i definitely wasn't expecting it to be changed. It's just something i noticed that we could potentially fix at some point. POST/PATCH being used incorrectly isn't the end of the world though.